File: web/src/lib/tenant.ts
import { api } from './api'

export interface TenantMember {
  employee_id: string
  email: string
  full_name: string
  /** Lower-cased local-part of email — подпись в попапе и legacy-токен. */
  handle: string
  /** Готовый токен для вставки: `Имя_Фамилия` либо логин. Считает сервер. */
  mention: string
}

export const tenantApi = {
  members: (q?: string, limit = 10): Promise<TenantMember[]> =>
    api
      .get<TenantMember[]>('/tenant/members', { params: { q: q || undefined, limit } })
      .then((r) => r.data),
}

File: app/schemas/tenant.py
"""Tenant-wide reference schemas (members for @mention autocomplete, etc.)."""

from __future__ import annotations

from uuid import UUID

from pydantic import BaseModel, ConfigDict


class TenantMemberBrief(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    employee_id: UUID
    email: str
    full_name: str
    # Local-part of email — convenience for client-side @mention rendering.
    handle: str
